src/Microsoft.AspNetCore.Mvc.RazorPages/Internal/CompiledPageRouteModelProvider.cs

```diff
@@ -17,14 +19,24 @@ public class CompiledPageRouteModelProvider : IPageRouteModelProvider
         private readonly object _cacheLock = new object();
         private readonly ApplicationPartManager _applicationManager;
         private readonly RazorPagesOptions _pagesOptions;
+        private readonly ILogger<CompiledPageRouteModelProvider> _logger;
         private List<PageRouteModel> _cachedModels;
 
         public CompiledPageRouteModelProvider(
```

Comment:
Do we need this? Is this just for unit tests? Consider just nuking it. It's easy in tests to just use the null logger